### PR TITLE
Add data layouts documentation

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
- It adds documentation to TensorLayout and its usage in DALI

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Added expected input data layout information to each operator documentation*
 - Affected modules and functionalities:
     *Docs of all operators*
 - Key points relevant for the review:
     *Changes in the documentation*
 - Validation and testing:
     *Pre-existing tests*
 - Documentation (including examples):
     *TODO*

**JIRA TASK**: *[DALI-1278]*

### CODE REVIEW

File: dali/pipeline/operator/op_schema.h [hunk 1]
```diff
@@ -387,6 +388,10 @@ class DLL_PUBLIC OpSchema {
     }
   }
 
+  DLL_PUBLIC inline const std::vector<std::vector<TensorLayout>> &SupportedLayouts() const {
```
Comment (mzient) [inline dali/pipeline/operator/op_schema.h]: I think that it should be per-input, like GetInputDox and GetInputName.

Comment (jantonguirao) [reply dali/pipeline/operator/op_schema.h]: Done

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Pull request merged**